tests/Integration/Behaviour/Features/Scenario/Category/category_management.feature

```diff
@@ -1,165 +1,752 @@
+      | name[en-US]                   | Mobile phones super            |
+      | name[fr-FR]                   | Mobile phones super fr         |
+      | active                        | true                           |
+      | parent category               | home-accessories               |
+      | link rewrite[en-US]           | mobile-phones-super-en         |
+      | link rewrite[fr-FR]           | mobile-phones-super-fr         |
+      | group access                  | guestGroup                     |
+      | associated shops              | shop1                          |
+      | description[en-US]            | description english            |
+      | description[fr-FR]            | description french             |
+      | additional description[en-US] | additional description english |
+      | additional description[fr-FR] | additional description french  |
+      | meta description[en-US]       | meta description english       |
+      | meta description[fr-FR]       | meta description french        |
+      | meta title[en-US]             | meta title english             |
+      | meta title[fr-FR]             | meta title french              |
+      | meta keywords[en-US]          | meta,keyword,english           |
+      | meta keywords[fr-FR]          | meta,keyword,french            |
 
   Scenario: Delete category
-    When I delete category "category1" choosing mode "associate_and_disable"
-    Then category "category1" does not exist
+    Given I add new category "category3" with following details:
+      | name[en-US]         | Mobile phones3   |
+      | name[fr-FR]         | Mobile phones fr |
+      | active              | false            |
+      | parent category     | home             |
+      | link rewrite[en-US] | mobile-phones-en |
+      | link rewrite[fr-FR] | mobile-phones-fr |
+    And I add new category "category4" with following details:
+      | name[en-US]         | Mobile phones4   |
+      | name[fr-FR]         | Mobile phones fr |
+      | active              | false            |
+      | parent category     | home             |
+      | link rewrite[en-US] | mobile-phones-en |
+      | link rewrite[fr-FR] | mobile-phones-fr |
+    And I add new category "category5" with following details:
+      | name[en-US]         | Mobile phones5   |
+      | name[fr-FR]         | Mobile phones fr |
+      | active              | false            |
+      | parent category     | home             |
+      | link rewrite[en-US] | mobile-phones-en |
+      | link rewrite[fr-FR] | mobile-phones-fr |
+    When I delete category "category3" choosing mode "associate_and_disable"
+    Then category "category3" does not exist
+    When I delete category "category4" choosing mode "associate_only"
+    Then category "category4" does not exist
+    When I delete category "category5" choosing mode "remove_associated"
+    Then category "category5" does not exist
```

Comment:
It doesn't really test the effect of each mode, we'd need to associate products to check this But I think it should be handled in a dedicated PR and a dedicated feature file At least it validates that each mode is usable 😅

Reply:
yep this was just a basic test to see that all modes deletes the category. And as you've already seen bellow I implemented dedicated cases to test every method when the category is the last for the product.